tools/torch_quant/tests/test_module_fx_trace.py

```diff
@@ -44,14 +49,110 @@ def test_simple_module(self, module: torch.nn.Module, input_size: Tuple[int, ...
     def test_untracable(self, module: torch.nn.Module) -> None:
         self.assertRaises(TraceError, fx_trace, module)
 
-    @unittest.skip('not implemented')
-    def test_filter_include_name(self) -> None:
+    def test_filter_include_names(self) -> None:
         module = SimpleModule()
         mapping = fx_trace(
             module, module_filter=ModuleFilter(include_names=['conv']))
         self.assertIn('conv', mapping)
         self.assertEqual(len(mapping), 1)
-        self.assertIs(mapping['conv'], module.conv1)
+        self.assertIs(mapping['conv'].m, module.conv)
+
+    def test_filter_include_classes(self) -> None:
+        module = SimpleModule()
+        module_filter = ModuleFilter(include_classes=[SubModule])
+        mapping = fx_trace(module, module_filter=module_filter)
+        self.assertIn('sub', mapping)
+        self.assertEqual(len(mapping), 1)
+        self.assertIs(mapping['sub'].m, module.sub)
+
+    def _check_is_leaf_module(
+        self, name: str, module: torch.nn.Module, graph: torch.fx.Graph
+    ):
+        targets = [nd.target for nd in graph.nodes if nd.op == 'call_module']
+        self.assertIn(name, targets)
+        for m, _ in module.named_children():
+            self.assertNotIn('.'.join([name, m]), targets)
+
+    def test_filter_exclude_names(self) -> None:
+        module = SimpleModule()
+        module_filter = ModuleFilter(exclude_names=['sub'])
+        mapping = fx_trace(module, module_filter=module_filter)
+        self.assertIn('', mapping)
+        self.assertEqual(len(mapping), 1)
+        self._check_is_leaf_module('sub', module.sub, mapping[''].gm.graph)
+
+    def test_filter_exclude_classes(self) -> None:
+        module = UntraceableSimpleModule()
+        module_filter = ModuleFilter(exclude_classes=[UntraceableSubModule])
+        mapping = fx_trace(module, module_filter=module_filter)
+        self.assertIn('', mapping)
+        self.assertEqual(len(mapping), 1)
+        self._check_is_leaf_module(
+            'untraceable_sub', module.untraceable_sub, mapping[''].gm.graph
+        )
+
+    def _test_filter_include_names_with_exclude_module(
+        self, module_filter: ModuleFilter
+    ) -> None:
+        module = UntraceableSimpleModule()
+        mapping = fx_trace(module, module_filter=module_filter)
+        self.assertIn('traceable_sub', mapping)
+        self.assertIn('untraceable_sub.linear_relu', mapping)
+        self.assertEqual(len(mapping), 2)
+        self._check_is_leaf_module(
+            'sub', module.traceable_sub.sub, mapping['traceable_sub'].gm.graph
+        )
```

Comment:
`_test_filter_include_names_with_exclude_module` can be refactored as a test function using parameterized

Reply:
Done